src/utils/calculate.ts

```diff
@@ -0,0 +1,12 @@
+import { CartProduct } from '@/types/cart';
+
+export const calculateTotalQuantity = (cart: CartProduct[]) => {
+  return cart.reduce((acc, cur) => acc + (cur.quantity ?? 0), 0);
+};
+
+export const calculateTotalAmount = (cart: CartProduct[]) => {
+  return cart.reduce(
+    (acc, cur) => acc + (cur.quantity ?? 0) * cur.product.price,
+    0
+  );
+};
```

Comment:
지금도 훌륭한 util이지만 한 가지 제안만 드려봅니다~
util 모듈의 이름이 'calculate'인 만큼 특정 도메인의 타입을 알기보다는 더 일반적인 유틸이 작성되면 좋을 듯해요!
필요한 interface를 여기서 작성하고 사용처에서 그 Interface에 맞는 인수를 제공하도록 하면 더 일반적인 유틸로 사용할 수 있을 듯 합니다! 그렇게 까지 할 필요가 안느껴진다면 /utils/cart 하위로 해당 함수들을 옮겨도 좋겠습니다!

```suggestion
export const calculateTotalQuantity = (items: { quantity: number }[]) => {
  return items.reduce((acc, item) => acc + item.quantity, 0);
};

export const calculateTotalAmount = (items: { quantity: number, price: number }[]) => {
  return items.reduce(
    (acc, item) => acc + item.quantity * item.price,
    0
  );
};
```

Reply:
도메인을 걷어내고 일반적으로 작성하는 게 재사용 측면에서도 좋아 보이네요!